src/Engine/Segments/SegmentConditionModel.php

```diff
@@ -119,4 +128,40 @@ public function matchesTraitValue($traitValue): bool
 
         return $condition;
     }
+
+    private function matchesSemverTraitValue($trait, $value, $condition)
+    {
+        switch ($this->operator) {
+            case (SegmentConditions::EQUAL):
+                $condition = version_compare($trait, $value, '==');
+                break;
+            case (SegmentConditions::GREATER_THAN):
+                $condition = version_compare($trait, $value, '>');
+                break;
+            case (SegmentConditions::GREATER_THAN_INCLUSIVE):
+                $condition = version_compare($trait, $value, '>=');
+                break;
+            case (SegmentConditions::LESS_THAN):
+                $condition = version_compare($trait, $value, '<');
+                break;
+            case (SegmentConditions::LESS_THAN_INCLUSIVE):
+                $condition = version_compare($trait, $value, '<=');
+                break;
+            case (SegmentConditions::NOT_EQUAL):
+                $condition = version_compare($trait, $value, '!=');
+                break;
+            case (SegmentConditions::CONTAINS):
```

Comment:
I think we can safely ignore the following 3 cases. It doesn't make sense to use contains, not contains or regex operators for semver values.